drivers/adc/Kconfig.nrfx

```diff
@@ -29,3 +29,16 @@ config ADC_NRFX_SAADC
 	select ADC_CONFIGURABLE_INPUTS
 	help
 	  Enable support for nrfx SAADC driver for nRF52 MCU series.
+
+# Workaround for not being able to have commas in macro arguments
+DT_COMPAT_NORDIC_NRF_ADC   := nordic,nrf-adc
+DT_COMPAT_NORDIC_NRF_SAADC := nordic,nrf-saadc
+
+# We enable ADC_0 here for the sake of other code in Zephyr that
+# depends on it. Device instantiation within nRF ADC drivers is
+# entirely driven by whether the devicetree nodes with compatibles
+# nordic,nrf-adc and nordic,nrf-saadc are enabled.
+config ADC_0
+	bool
+	default y if ADC_NRFX_ADC && $(dt_nodelabel_has_compat,adc,$(DT_COMPAT_NORDIC_NRF_ADC))
```

Comment:
Perhaps it would be better to make `ADC_NRFX_ADC` dependent on `$(dt_nodelabel_has_compat,adc,$(DT_COMPAT_NORDIC_NRF_ADC))` and do the same with SAADC (both these drivers won't compile successfully without the corresponding node enabled). Then you could use just `default ADC_NRFX_ADC || ADC_NRFX_SAADC` here.